examples/pysdl2.py

```diff
@@ -0,0 +1,218 @@
+            
+    def GetViewRect(self, rect_out, **_):
+        rect_out.extend([0, 0, self.__width, self.__height])
+        return True
+    
+    def OnPaint(self, browser, element_type, paint_buffer, **_):
+        """
+        Using the pixel data from CEF's offscreen rendering
+        the data is converted by PIL into a SDL2 surface
+        which can then be rendered as a SDL2 texture.
+        """
+        if element_type == cef.PET_VIEW:
+            image = Image.frombuffer(
+                'RGBA',
+                (self.__width, self.__height),
+                paint_buffer.GetString(mode="rgba", origin="top-left"),
+                'raw',
+                'BGRA'
+            )
+            #
+            # Following PIL to SDL2 surface code from pysdl2 source.
+            #
+            mode = image.mode
+            rmask = gmask = bmask = amask = 0
+            if mode == "RGB":
+                # 3x8-bit, 24bpp
+                if sdl2.endian.SDL_BYTEORDER == sdl2.endian.SDL_LIL_ENDIAN:
+                    rmask = 0x0000FF
+                    gmask = 0x00FF00
+                    bmask = 0xFF0000
+                else:
+                    rmask = 0xFF0000
+                    gmask = 0x00FF00
+                    bmask = 0x0000FF
+                depth = 24
+                pitch = self.__width * 3
+            elif mode in ("RGBA", "RGBX"):
+                # RGBX: 4x8-bit, no alpha
+                # RGBA: 4x8-bit, alpha
+                if sdl2.endian.SDL_BYTEORDER == sdl2.endian.SDL_LIL_ENDIAN:
+                    rmask = 0x00000000
+                    gmask = 0x0000FF00
+                    bmask = 0x00FF0000
+                    if mode == "RGBA":
+                        amask = 0xFF000000
+                else:
+                    rmask = 0xFF000000
+                    gmask = 0x00FF0000
+                    bmask = 0x0000FF00
+                    if mode == "RGBA":
+                        amask = 0x000000FF
+                depth = 32
+                pitch = self.__width * 4
+            else:
+                print "Unsupported mode: %s" % mode
+                exit_app()
+            
+            pxbuf = image.tobytes()
+            # create surface
+            surface = sdl2.SDL_CreateRGBSurfaceFrom(pxbuf, self.__width, self.__height, depth, pitch, rmask, gmask, bmask, amask)
+            if self.texture:
+                # free memory used by previous texture
+                sdl2.SDL_DestroyTexture(self.texture)
+            # create texture
+            self.texture = sdl2.SDL_CreateTextureFromSurface(self.__renderer, surface)
+            # free the surface
+            sdl2.SDL_FreeSurface(surface)
+        else:
+            print "Unsupport element_type in OnPaint"
+
+def exit_app():
+    """Tidy up SDL2 and CEF before exiting."""
+    sdl2.SDL_Quit()
+    cef.Shutdown()
+    print "exited"
+
+def main():
+    # the following variables control the dimensions of the window
```

Comment:
Comments should start from upper case letter - there are more fixes like this required.